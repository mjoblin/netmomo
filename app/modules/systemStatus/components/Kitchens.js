import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'antd';


export const Kitchens = ({ kitchens }) => {
    const columns = [
        {
            title: 'Name',
            dataIndex: 'kitchen_name',
        },
        {
            title: 'Host',
            dataIndex: 'host',
        },
        {
            title: 'Port',
            dataIndex: 'port',
        },
        {
            title: 'Interface',
            dataIndex: 'interface',
        },
        {
            title: 'Filter',
            dataIndex: 'filter',
        },
        {
            title: 'Poke (secs)',
            dataIndex: 'poke_interval',
        },
        {
            title: 'Chefs',
            dataIndex: 'chefs',
            render: chefs => chefs.map(
                chef => <span key={chef}>{chef}<br /></span>),
        },
    ];

    return (
        <Table
            dataSource={kitchens}
            columns={columns}
            pagination={false}
            rowKey={row => `${row.kitchen_name}_${row.host}_${row.port}`}
            size="middle"
        />
    );
};


Kitchens.propTypes = {
    kitchens: PropTypes.array.isRequired,
};

export default Kitchens;

