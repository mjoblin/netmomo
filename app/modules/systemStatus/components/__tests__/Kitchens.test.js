import React from 'react';
import { shallow } from 'enzyme';
import { Kitchens } from '../Kitchens';
import testData from './testData.json';

describe('Eaters', () => {
    it('matches snapshot', () => {
        const component = shallow(
            <Kitchens
                kitchens={testData.dumpling_kitchens}
            />
        );

        expect(component).toMatchSnapshot();
    });
});