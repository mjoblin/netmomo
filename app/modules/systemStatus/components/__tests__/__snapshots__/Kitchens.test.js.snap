// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Eaters matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Kitchens
    kitchens={
      Array [
        Object {
          "chefs": Array [
            "netdumplings.dumplingchefs.ARPChef",
            "netdumplings.dumplingchefs.DNSLookupChef",
            "netdumplings.dumplingchefs.PacketCountChef",
          ],
          "filter": "tcp or icmp or udp or arp",
          "host": "127.0.0.1",
          "interface": "en0",
          "kitchen_name": "default_kitchen",
          "poke_interval": 5,
          "port": 53012,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "bordered": false,
      "className": "",
      "columns": Array [
        Object {
          "dataIndex": "kitchen_name",
          "title": "Name",
        },
        Object {
          "dataIndex": "host",
          "title": "Host",
        },
        Object {
          "dataIndex": "port",
          "title": "Port",
        },
        Object {
          "dataIndex": "interface",
          "title": "Interface",
        },
        Object {
          "dataIndex": "filter",
          "title": "Filter",
        },
        Object {
          "dataIndex": "poke_interval",
          "title": "Poke (secs)",
        },
        Object {
          "dataIndex": "chefs",
          "render": [Function],
          "title": "Chefs",
        },
      ],
      "dataSource": Array [
        Object {
          "chefs": Array [
            "netdumplings.dumplingchefs.ARPChef",
            "netdumplings.dumplingchefs.DNSLookupChef",
            "netdumplings.dumplingchefs.PacketCountChef",
          ],
          "filter": "tcp or icmp or udp or arp",
          "host": "127.0.0.1",
          "interface": "en0",
          "kitchen_name": "default_kitchen",
          "poke_interval": 5,
          "port": 53012,
        },
      ],
      "indentSize": 20,
      "loading": false,
      "locale": Object {},
      "pagination": false,
      "prefixCls": "ant-table",
      "rowKey": [Function],
      "rowSelection": null,
      "showHeader": true,
      "size": "middle",
      "useFixedHeader": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "bordered": false,
        "className": "",
        "columns": Array [
          Object {
            "dataIndex": "kitchen_name",
            "title": "Name",
          },
          Object {
            "dataIndex": "host",
            "title": "Host",
          },
          Object {
            "dataIndex": "port",
            "title": "Port",
          },
          Object {
            "dataIndex": "interface",
            "title": "Interface",
          },
          Object {
            "dataIndex": "filter",
            "title": "Filter",
          },
          Object {
            "dataIndex": "poke_interval",
            "title": "Poke (secs)",
          },
          Object {
            "dataIndex": "chefs",
            "render": [Function],
            "title": "Chefs",
          },
        ],
        "dataSource": Array [
          Object {
            "chefs": Array [
              "netdumplings.dumplingchefs.ARPChef",
              "netdumplings.dumplingchefs.DNSLookupChef",
              "netdumplings.dumplingchefs.PacketCountChef",
            ],
            "filter": "tcp or icmp or udp or arp",
            "host": "127.0.0.1",
            "interface": "en0",
            "kitchen_name": "default_kitchen",
            "poke_interval": 5,
            "port": 53012,
          },
        ],
        "indentSize": 20,
        "loading": false,
        "locale": Object {},
        "pagination": false,
        "prefixCls": "ant-table",
        "rowKey": [Function],
        "rowSelection": null,
        "showHeader": true,
        "size": "middle",
        "useFixedHeader": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
